### DISCUSSION TITLE
Use Cases

### DISCUSSION THREAD

**Discussion by FabrizioMoggio** [camaraproject/CallForwardingSignal#3] [Ideas]
In the API description, concrete use cases must be provided. The use cases can be Access specific.

The idea is to add use cases here and discuss about them on specific Discussions. We can open here a new use case as "UseCase-x: use case name" with a description on the use case. Anyone that want to discuss about it can open a new Discussion with "UseCase-x: use case name" in the title. So we have here a clean list of the uses cases.

**Comment by FabrizioMoggio** [comment]
**UseCase-1: Bank Frauds** 

Banks offer the option for users to receive telephone calls to solve problems, e.g. in case of issues with the bank App, issues with Smartphones, the bank website or any tool providing access to the online bank account management system.
How does the fraudster operate?
• Through social engineering, the fraudster gathers information about bank customers who are engaged in such a situation, needing the bank to contact them.
• Once they identify the service provider and phone number of a customer, they manage to successfully activate, on the customer profile, the 'unconditional call forwarding' to a phone number they control.
• After successfully setting up 'unconditional call forwarding,' the fraudster then contacts the bank, claiming an inability to access online banking services and requesting a call from a bank representative.
• When the bank calls the registered contact number of the user, the active 'unconditional call forwarding' diverts the call directly to the fraudster, thus completing their scheme.
The Call Forwarding Signal API can be used by the bank to verify if a "unconditional call forwarding" service is active on the customer's phone, avoiding such a fraud.

**Intents**:

- Intent 1: https://github.com/camaraproject/CallForwardingSignal/discussions/6#discussioncomment-8679641

**Discussion**: NA

**User Story**: https://github.com/camaraproject/CallForwardingSignal/blob/main/documentation/API_documentation/Call-Forwarding-Signal-UserStory-UseCase-1.md

**Status**: Supported by the API (v0.1.4-wip). Included in Fall 24 Meta Release

**Comment by chinaunicomyangfan** [comment]
**UseCase-2: Alert Interception**

The call forwarding function can be used by telecommunications fraudsters to prevent victims from receiving timely reminders. Due to fraudsters controlling the victim's phone and enabling unconditional call forwarding, police or family members can only contact the wrong number when contacting the victim. The call forwarding signal API can help the police identify whether the number has set up unconditional call forwarding, so as to quickly choose other ways to contact the victim, such as instant messaging software. 

**Intents**: 

- Intent 1: https://github.com/camaraproject/CallForwardingSignal/discussions/6#discussioncomment-8679641

**Discussion**: https://github.com/camaraproject/CallForwardingSignal/discussions/9

**User Story**: NA

**Status**: Supported by the API (v0.1.4-wip). Included in Fall 24 Meta Release

**Comment by enlaichu** [comment]
**UseCase-3: Call FWD Verification**

Conditional call forwarding can be used by users to forward calls to robocall detection services (eg Nomorobo) or enhanced voicemail services (Eg Youmail) or virtual assistant service (Productive.ai) who  can answer the call if the original user is not available to answer, providing a better level of customer service to callers than the standard carrier voicemail. 

When users are asked to conditionally forward their number by tapping a forwarding code (eg `*004*<forwarding number>#`), this conditional forwarding does not always work and requires users to retry the code or restart their phone and try again.  

There is no easy way for the application to programatically detect if the forwarding rule has been applied correctly or if it is still active (the user sometimes resets their network accidentally or if they travel and come back in the country, or if they switch phones or SIM cards). 

Hence it would be helpful for the application to ensure the forwarding rule is active without the user having to find another phone to call him/herself to test. 

**Intents**: 

- Intent 2: https://github.com/camaraproject/CallForwardingSignal/discussions/6#discussioncomment-9327549

**Discussion**: https://github.com/camaraproject/CallForwardingSignal/discussions/17

**User Story**: NA

**Status**:  supported by the API (v 02.0-wip). Included in Fall 24 Meta Release
